### PR TITLE
refactor action and replace team id by user id

### PR DESCRIPTION
**Pull request by branquinhoaa**
PR related to the issue #826

- Use user id instead teams id into conference gid generator.
- Redirect to conferences after orga create new conference ( but to edit/teams when is a student that create a new conference )
- Refactor action

### CODE REVIEW

File: app/controllers/conferences_controller.rb [hunk 1]
```diff
@@ -33,8 +32,15 @@ def redirect
 
   private
 
-  def generate_gid(team)
-    "#{Season.current.name}-#{Time.now.getutc.to_i}-#{team.id}"
+  def build_conference
+    conference = Conference.new(conference_params)
+    conference.season_id = current_season.id
```
Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: You can set the season_id at the same time as the conference_params like this:
`conference = Conference.new(conference_params.merge(season: current_season))`
The same goes (I think) for the generated `gid`, but I don't mind if you leave that separately.

File: app/controllers/conferences_controller.rb [hunk 2]
```diff
@@ -33,8 +32,15 @@ def redirect
...
+    conference.gid = generate_gid
+    conference
+  end
+
+  def generate_gid
+    "#{Season.current.name}-#{Time.now.getutc.to_i}-#{current_user.id}"
```
Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: My main question: If I understand this correctly, this means that _every_ logged_in user can add conferences to the main list now? Because this controller is about the conferences that are imported via the importer, right?

File: app/controllers/conferences_controller.rb [hunk 3]
```diff
@@ -16,12 +16,11 @@ def show
 
   def create
     team = current_user.student_team
```
Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: What happens here if the current_user is not a student?

File: app/controllers/conferences_controller.rb (hunk 2 above)
Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: Can you add tests please for the non-student users? The current specs cover only logged in _students_ , no other users.
You can wait to do them untill we finished the Q&A ;-)

File: app/controllers/conferences_controller.rb (hunk 3 above)
Comment (branquinhoaa) [inline app/controllers/conferences_controller.rb]: if a current user is not a student, the team will be empty. 
When the conference was saved, the 'team.present?' will return false and then the user will be redirected to a conferences_path after creating a new conference.
However, this button just appears for users that have an ability to 'update_conference_preferences' -  just orga members and students of that specific team.

File: app/controllers/conferences_controller.rb (hunk 1 above)
Comment (branquinhoaa) [inline app/controllers/conferences_controller.rb]: Good idea! :dancer:

File: app/controllers/conferences_controller.rb (hunk 2 above)
Comment (branquinhoaa) [inline app/controllers/conferences_controller.rb]: The button to add new conference are supposed to appear just for a student of a specific team and orga members. However, orga members do not have a team, so I thought that use the current user id in the GID could help to track who added the conference, instead what team added the conference. But as far as I know, just the students (and orga members) are allowed to add conferences. You are right, I have to include tests for it! :D thank you

File: app/controllers/conferences_controller.rb [hunk 4]
```diff
@@ -33,10 +33,7 @@ def redirect
   private
 
   def build_conference
-    conference = Conference.new(conference_params)
-    conference.season_id = current_season.id
-    conference.gid = generate_gid
-    conference
+    Conference.new(conference_params.merge(season: current_season, gid: generate_gid))
```
Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: 👍 Nice!

File: app/controllers/conferences_controller.rb (hunk 3 above)
Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: Aha, it is only for orga members and students. Then I still don't get the use case for the change to user_id. Sorry ;-) This controllers#create is were the students enter their 'suggestions', right?
So, why should an orga member use _this_ create action?  Why not either in the namespaced orga/conferences? OR even better, import them via the google database. 
I have the feeling that we are trying to press two very different use cases into one. WDYT?

**Comment by branquinhoaa** [comment]
This one is related to the issue #826 (fix a bug in button to a new conference in teams/edit ). The other issue (#825) is an enhancement  - just to add an extra button. 
But if you want I can join in one PR.

File: app/controllers/conferences_controller.rb (hunk 3 above)
Comment (branquinhoaa) [inline app/controllers/conferences_controller.rb]: Yeah, makes sense just team's student create new conferences. However, in the description of this issue, it was asked to let orga members do it too. Even in the new issue #825 related to add new button, was asked to allow both orga members and students to add new conference.

Comment (F3PiX) [inline app/controllers/conferences_controller.rb]: Okay, I see what you mean! Do you mind if we set this on hold, then? And leave it in this separate PR. 
I am sorry for the complications :-(
But let's first _implement_ the enhancement before we solve bugs in that enhancement 😂

**Comment by F3PiX** [comment]
Hey @branquinhoaa 👋  
Now that #829 has been merged, would you mind rebasing?

**Comment by klappradla** [comment]
Could you please bring this up 2 date with `master` @branquinhoaa ? I think some changes in the `ConferencesController` are already obsolete.... 🙈 

Are all your changes in @F3PiX ?

**Comment by F3PiX** [comment]
@klappradla @branquinhoaa Looks like we are cross-typing ;-)
I'll update/sync my review when the branch is updated. 👍

**Comment by branquinhoaa** [comment]
@klappradla @F3PiX  I updated the PR.

**Comment by klappradla** [comment]
Yay, looks all good from my side. Can you check back @F3PiX ? I'd like to merge this soon.

**Comment by F3PiX** [comment]
Looks good to me too!! Thanks @branquinhoaa 🥇

**Comment by mkalininait** [comment]
Still getting an error when trying to create a new conf as an orga :/
